src/levanter/data/text.py

```diff
@@ -584,6 +588,103 @@ def tagged_eval_sets(
         return [(eval_sets[name], tags[name]) for name in eval_sets]
 
 
+@dataclass
+class LMSupervisedDatasetConfig(LMDatasetSourceConfig):
+    """This class represents a dataset source with URLs or hf name/id."""
+
+    cache_dir: str = "cache/"
+
+    tags: Optional[List[str]] = None
+    """tags for the dataset. Typically the name of the dataset in the config will be added as a tag as well"""
+    name: Optional[str] = None  # name for hf dataset
+
+    validation_urls: List[str] = ()  # type:ignore
+
+    # def token_seq_dataset(
```

Comment:
Can you remove this code if it's commented out?

Reply:
It was throwing an error from the precommit. I need to check with David if we need these functions. If so, I'll need to fix some of these

Reply:
rm and don't have LMSupervisedDatasetConfig implement LMDatasetSourceConfig